image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl

```diff
@@ -252,6 +252,20 @@
 
 {{ include "srox.initGlobalPrefix" (list $) }}
 
+{{/*
+The following block checks for the validity of the provided init bundle. (`helm install ... -f <init-bundle.yaml>`)
+1. Throw an error if no init bundle was provided
+2. Throw an error if the wrong init-bundle format was provided (user wrongly passed an operator-formatted init-bundle to helm). We detect the presence of an operator init bundle by checking for the Kubernetes secret annotation "init-bundle.stackrox.io/name", that suggests a user has wrongly used `helm install ... -f <operator-init-bundle.yaml>.`
+3. Warn if the user specified both init bundles, e.g. `helm install ... -f <helm-init-bundle.yaml> -f <operator-init-bundle.yaml>`
+*/}}
+{{- if and (hasKey (._rox.metadata).annotations "init-bundle.stackrox.io/name") (empty ._rox.ca._cert) }}
```

Comment:
It's always good to try writing the code in such a way that it is easy to grasp and understand. Reading long conditions is a bit tricky, what do you think about doing something like:

```
{{ $operatorInitBundlePresent := hasKey (.Values.metadata).annotations "init-bundle.stackrox.io/name" }}
{{ $helmInitBundlePresent := not (empty ._rox.ca._cert) }}
```

then you can write your conditions in a way which is much easier for future readers of the code to understand:

```
{{- if $helmInitBundlePresent }}
  {{- if $operatorInitBundlePresent }}
    ... /* Emit a warning along the lines of: ignoring operator init bundle. */
  {{- end }}
{{- else }}
  {{- if $operatorInitBundlePresent }}
    ... /* Fail with message along the lines: The init bundle was provided in the wrong format... please provide the right one instead. */
  {{- else }}
    ... /* Fail with message along the lines: No init bundle provided, please provide the Helm chart init bundle. */
  {{- end }}
{{- end }}
```